File: Blackjack.py
import random

class Blackjack(object):
    def __init__(self):
        print('Weclome to The Blackjack game')
        self.balance = int(input('Enter how much you want to play?'))
        self.bit = int(input('How much token you want to bit? 100 or 500 or 1000:'))


    def game_card(self):
        player_card1 = random.randint(1, 11)
        player_card2 = random.randint(1, 11)
        computer_card1 = random.randint(1, 11)
        computer_card2 = random.randint(1, 11)
        self.computer_card = computer_card1 + computer_card2
        self.player_card = player_card1 + player_card2
        print(f'Your card: {player_card1},{player_card2} \nYour competitor card: {computer_card1}')
        answer = input('Hit or Stand?:')
        if answer == 'Hit':
            self.player_card  += random.randint(1, 11)
            print(self.player_card)
        else:
            print(self.player_card)

    def win_conuation(self):
        win = []
        if (self.player_card > 21 and self.computer_card > 21) or (21 - self.player_card < 21 - self.computer_card) or (self.player_card == 21):
            win == 'Player win'
            self.balance += self.bit
            print(f'You win! Your have {self.player_card} and Your competitor only have {self.computer_card} \nYou have {self.balance}')
        elif (self.computer_card > 21 and self.player_card > 21) or (21 - self.computer_card < 21 - self.player_card) or (self.computer_card == 21):
            win == 'Computer win'
            self.balance -= self.bit
            print(f'You lostt! Your only have {self.player_card} and Your competitor  have {self.computer_card} \nYou have {self.balance}')
        elif (self.player_card == self.computer_card):
            win =='Push'
            return win
        elif self.balance < self.bit:
            print('Sorry, you lose all the money')

    def replay(self):
        return input('Do you want to play again? Y or N').startswith('Y')

def main():
        while True:
            card_game = Blackjack()
            card_game.game_card()
            card_game.win_conuation()
            if card_game.replay() == False:
                break


if __name__ == "__main__":
    main()

File: Blackjack.java
import java.util.Scanner;
import java.util.Random;
import java.lang.Math;
import java.util.*;

public class Blackjack {

    static int player_card1;
    static int player_card2;
    static int computer_card1;
    static int computer_card2;
    static int player_card;
    static int computer_card;

    Blackjack() {
        Random random = new Random();

        // Using Random class in Java to add random number
        player_card1 = random.nextInt(11);
        player_card1 += 1;

        // Using Math.random() to add random number
        player_card2 = (int) (Math.random() * 11 + 1);
        computer_card1 = (int) (Math.random() * 11 + 1);
        computer_card2 = (int) (Math.random() * 11 + 1);
        computer_card = computer_card1 + computer_card2;
        player_card = player_card1 + player_card2;

    }

    public static void game_card(int player_card1, int player_card2, int player_card, int computer_card1,
            int computer_card2, int computer_card) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Hit or Stand?: ");
        String answer = scanner.nextLine();
        int another_card = (int) (Math.random() * 11 + 1);
        if (answer.equals("Hit")) {
            player_card += another_card;
            System.out.println("You have: " + another_card);
            System.out.println("Total You have " + player_card);
            System.out.print("The computer have: " + computer_card);
        } else {
            System.out.println("You have " + player_card);
            System.out.println("The computer have: " + computer_card);
        }
    }

    public static String getwinconuation() {
        String win = "";
        if ((player_card <= 21 && computer_card > 21) || (21 - player_card < 21 - computer_card)) {
            win = "Player win";
            System.out.println(win);
        } else if ((computer_card <= 21 && player_card > 21) || (21 - computer_card < 21 - player_card)) {
            win = "Computer win";
            System.out.println(win);
        } else if (player_card == computer_card) {
            win = "Push";
            System.out.println(win);
        }
        return win;
    }

    public static boolean replay() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Do you want to play again? Y or N: ");
        String userinput = scanner.nextLine();
        char letter = userinput.charAt(0);
        if (letter == 'Y' || letter == 'y') {
            return true;
        } else {
            return false;
        }

    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Blackjack play1 = new Blackjack();

        System.out.println("Weclome to The Blackjack game");
        do {
            System.out.println("Your card:" + player_card1 + " " + player_card2);
            System.out.println("The computer card:" + computer_card1);
            play1.game_card(player_card1, player_card2, player_card, computer_card1, computer_card2, computer_card);
            System.out.println(" ");
            play1.getwinconuation();
        } while (replay());
    }
}
